include/mlir/ExecutionEngine/OptUtils.h

```diff
@@ -41,9 +42,12 @@ void initializeLLVMPasses();
 
 /// Create a module transformer function for MLIR ExecutionEngine that runs
 /// LLVM IR passes corresponding to the given speed and size optimization
-/// levels (e.g. -O2 or -Os).
+/// levels (e.g. -O2 or -Os). If not null, \p targetMachine is used to
+/// initialize passes that provide target-specific information to the LLVM
+/// optimizer.
```

Comment:
Can you mention that the the provided TM is expected to outlive the returned std::function?